noxfile.py

```diff
@@ -40,10 +40,59 @@ def set_environment_variables(env_dict, session):
 def run_pybamm_requires(session):
     """Download, compile, and install the build-time requirements for Linux and macOS: the SuiteSparse and SUNDIALS libraries."""  # noqa: E501
     set_environment_variables(PYBAMM_ENV, session=session)
+    force_rebuild = "--force" in session.posargs
+    sundials_so_path = [
+            Path("/path/to/sundials/libsundials_idas.so"),
+            Path("/path/to/sundials/libsundials_idas.dylib"),
+
+            Path("/path/to/sundials/libsundials_sunlinsolklu.so"),
+            Path("/path/to/sundials/libsundials_sunlinsolklu.dylib"),
+
+            Path("/path/to/sundials/libsundials_sunlinsoldense.so"),
+            Path("/path/to/sundials/libsundials_sunlinsoldense.dylib"),
+
+            Path("/path/to/sundials/libsundials_sunlinsolspbcgs.so"),
+            Path("/path/to/sundials/libsundials_sunlinsolspbcgs.dylib"),
+
+            Path("/path/to/sundials/libsundials_sunlinsollapackdense.so"),
+            Path("/path/to/sundials/libsundials_sunlinsollapackdense.dylib"),
+
+            Path("/path/to/sundials/libsundials_sunmatrixsparse.so"),
+            Path("/path/to/sundials/libsundials_sunmatrixsparse.dylib"),
+
+            Path("/path/to/sundials/libsundials_nvecserial.so"),
+            Path("/path/to/sundials/libsundials_nvecserial.dylib"),
+
+            Path("/path/to/sundials/libsundials_nvecopenmp.so"),
+            Path("/path/to/sundials/libsundials_nvecopenmp.dylib")
+        ]
```

Comment:
The path to these `.so` and `.dylib` files is `$HOME/local/lib`, i.e., `LD_LIBRARY_PATH`  (we already have these variables set above as constants in the file which you can reuse).

Here, `.dylib` files exist only on macOS and similarly `.so` files only on Linux, so you should use `sys.platform` like it has been used in other places in the file to check the target platform. Also, could you write the filenames in the list first and then append the `.dylib` and `.so` extensions and construct `Path` objects later? The code will be much cleaner that way. Something like this:

```python

SUNDIALS_LIBS = [
    ...
    "libsundials_idas",
    "libsundials_nvecopenmp",
    ...
]
if sys.platform == "darwin"
    fileext = ".dylib"
if sys.platform == "linux"
    fileext = ".so"
```

and join filenames with `fileext`, create `Path` objects by looping over them, check if they exist in `LD_LIBRARY_PATH`. Alternatively, the `os` module can be used to check if these files exist (`os.path.isfile`), but this is less preferable because we already have a `pathlib` import.